crates/apollo-compiler/src/diagnostics.rs

```diff
@@ -265,6 +265,20 @@ pub enum DiagnosticData {
         /// Name of the field
         field: String,
     },
+    #[error("`{name}` field of type '{ty}' must not have a sub selection")]
+    NoSubselectionAllowed {
+        // field name
+        name: String,
+        // field type
+        ty: String,
+    },
+    #[error("`{name}` field of type '{ty}' must have a sub selection")]
+    MandatorySubselection {
```

Comment:
```suggestion
    MissingSubelection {
```
`rustc` diagnostics have a whole category of `Missing` errors when something that should be there isn't. @goto-bus-stop what are your thoughts?

Reply:
https://github.com/apollographql/apollo-rs/pull/465/commits/3c531a8a5151aeb4a6d356f95ec61d0f46583c41